### PR TITLE
CP-3571 App Version filter Feature added

### PR DESCRIPTION
**Pull request by jprajapati1996**
- Multiple screen feature
- Enable / Disable multiple screen
- Enable / Disable carousel

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java [hunk 1]
```diff
@@ -175,11 +168,12 @@ public void dismiss() {
 
     }
 
-    public void moveToNextScreen() {
-        int currentPosition = viewPager2.getCurrentItem();
-        if (currentPosition < data.getScreens().size() - 1) {
-            viewPager2.setCurrentItem(currentPosition + 1);
-        }
+    public void moveToNextScreen(int i) {
+        viewPager2.setCurrentItem(i, true); //getItem(-1) for previous
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:172]: remove comment

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java [hunk 2]
```diff
@@ -175,11 +168,12 @@ public void dismiss() {
 
     }
 
-    public void moveToNextScreen() {
-        int currentPosition = viewPager2.getCurrentItem();
-        if (currentPosition < data.getScreens().size() - 1) {
-            viewPager2.setCurrentItem(currentPosition + 1);
-        }
+    public void moveToNextScreen(int i) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:171]: rename `i` to something more descriptive

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java [hunk 3]
```diff
@@ -175,11 +168,12 @@ public void dismiss() {
...
+        viewPager2.setCurrentItem(i, true); //getItem(-1) for previous
+    }
+
+    private int getItem(int i) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:175]: rename `i` to something more descriptive

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java [hunk 4]
```diff
@@ -347,19 +341,16 @@ private void displayBanner(LinearLayout body) {
...
         if (data.isCloseButtonEnabled()) {
             buttonClose.setVisibility(View.VISIBLE);
-            buttonClose.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View view) {
-                    dismiss();
-                }
-            });
+            buttonClose.setOnClickListener(view -> dismiss());
+            Log.e(TAG, "CloseIcon: " + data.isCloseButtonEnabled());
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:353]: Remove log if not needed

File: cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java [hunk 5]
```diff
@@ -168,6 +172,12 @@ public String getPositionType() {
         return positionType;
     }
 
+
+    public boolean getEnableMultipleScreens() {
+        return enableMultipleScreens;
+    }
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java:179]: remove 1 line

File: cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java [hunk 6]
```diff
@@ -168,6 +172,12 @@ public String getPositionType() {
         return positionType;
     }
 
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java:175]: remove 1 line

File: cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java [hunk 7]
```diff
@@ -200,6 +210,7 @@ public List<HashMap<String, String>> getAttributes() {
         return attributes;
     }
 
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java:213]: remove 1 line

File: cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java [hunk 8]
```diff
@@ -341,6 +363,7 @@ public static Banner create(JSONObject json) throws JSONException {
             }
         }
 
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java:366]: remove 1 line

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlock.java [hunk 9]
```diff
@@ -1,18 +1,26 @@
...
 
-    public BannerBlockType getType() { return type; }
+    public BannerBlockType getType() {
+        return type;
+    }
 
     public static BannerBlock create(JSONObject json) throws JSONException, IllegalArgumentException {
         BannerBlockType blockType = BannerBlockType.fromString(json.getString("type"));
-
+        Log.e("Tag 1",""+json);
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlock.java:23]: remove log

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlockScreen.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+    public static BannerBlockScreen create(JSONObject json) throws JSONException {
+        BannerBlockScreen bannerBlockScreen = new BannerBlockScreen();
+        if (json != null) {
+            bannerBlockScreen.label = json.getString("label");
+            bannerBlockScreen.value = json.getString("value");
+        }
+        return bannerBlockScreen;
+    }
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlockScreen.java:30]: remove 1 line

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerImageBlock.java [hunk 11]
```diff
@@ -2,27 +2,44 @@
...
-    public boolean dismissOnClick() { return dismiss; }
+    public boolean dismissOnClick() {
+        return dismiss;
+    }
 
     public BannerAction getAction() {
         return action;
     }
 
+    public List<BannerBlockScreen> getBlocks() {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerImageBlock.java:38]: Is this needed?

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerButtonBlock.java [hunk 12]
```diff
@@ -13,23 +18,44 @@ public class BannerButtonBlock extends BannerBlock {
...
+        return radius;
+    }
 
-    public Alignment getAlignment() { return alignment; }
+    public BannerAction getAction() {
+        return action;
+    }
 
-    public int getRadius() { return radius; }
+    public List<BannerBlockScreen> getBlocks() {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerButtonBlock.java:55]: Is this needed?

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerTextBlock.java [hunk 13]
```diff
@@ -1,29 +1,49 @@
...
+    }
 
     public String getFamily() {
         return family;
     }
 
+    public List<BannerBlockScreen> getBlocks() {
+        return blockScreens;
+    }
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerTextBlock.java:44]: remove 2 empty lines

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerTextBlock.java (hunk 13 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerTextBlock.java:44]: okay i will

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerButtonBlock.java (hunk 12 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerButtonBlock.java:55]: yes

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerImageBlock.java (hunk 11 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerImageBlock.java:38]: yes

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlockScreen.java (hunk 10 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlockScreen.java:30]: Okay done

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java (hunk 1 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:172]: Yes i will removed

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java (hunk 4 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerPopup.java:353]: okay

File: cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java (hunk 5 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/Banner.java:179]: okay

File: cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlock.java (hunk 9 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/models/blocks/BannerBlock.java:23]: removed

**Pull request merged**